src/pool-bin/BinPositionManager.sol

```diff
@@ -203,13 +205,9 @@ contract BinPositionManager is
             ZERO_BYTES
         );
 
-        int256 amount0 = delta.amount0();
-        int256 amount1 = delta.amount1();
-        // delta amt0/amt1 will always be negative in mint case
-        if (amount0 > 0 || amount1 > 0) revert IncorrectOutputAmount();
-        if (uint128(uint256(-amount0)) < params.amount0Min || uint128(uint256(-amount1)) < params.amount1Min) {
-            revert OutputAmountSlippage();
-        }
+        /// Slippage checks, similar to CL type. However, this is different from TJ, in PCS v4,
+        /// as hooks can impact delta (take extra token), user need to be protected with amountMax instead
+        delta.validateMaxIn(params.amount0Max, params.amount0Max);
```

Comment:
key change in this PR where 

- before: `add_liquidity` user slippage protected with `amount0Min | amount1Min` 
- after: `add_liquidity` user slippage protected with `amount0Max | amount1Max` (align with CL)